drivers/frequency/adf4377/adf4377.c

```diff
@@ -755,6 +754,151 @@ int adf4377_get_rfout(struct adf4377_dev *dev, uint64_t *val)
 	return 0;
 }
 
+/**
+ * @brief Set the value of SR_DEL Adjustment Feature Translated to Pico Seconds
+ * on the output. Reset to Max  value of 127.
+ * @param dev 		- The device structure.
+ * @param val	 	- The desired SR_DEL register value.
+ * @return    		- Result of the writing procedure, error code otherwise.
+ */
+int adf4377_set_sr_del_adj(struct adf4377_dev *dev, int32_t val)
+{
+	uint8_t tmp;
+
+	if (!dev)
+		return -EINVAL;
+
+	dev->sr_del_adj = (uint8_t)val;
+
+	if (val > ADF4377_SR_DEL_MAX)
+		dev->sr_del_adj = ADF4377_SR_DEL_MAX;
+
+	tmp = ADF4377_SR_DEL(dev->sr_del_adj);
+	return adf4377_spi_update_bit(dev, 0x43, ADF4377_SR_DEL_MSK, tmp);
+}
+
+/**
+ * @brief Gets the value of the set SR_DEL Adjustment Feature Translated to Pico
+ * Seconds on the output.
+ * @param dev 		- The device structure.
+ * @param val	 	- The read SR_DEL register value.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4377_get_sr_del_adj(struct adf4377_dev *dev, int32_t *val)
+{
+	uint8_t tmp;
+	int ret;
+
+	ret = adf4377_spi_read(dev, 0x43, &tmp);
+	if (ret)
+		return ret;
+
+	dev->sr_del_adj = no_os_field_get(ADF4377_SR_DEL_MSK, tmp);
+	*val = dev->sr_del_adj;
+
+	return 0;
+}
+
+/**
+ * @brief Set the value of SR_INV_ADJ Adjustment to enable or disable which adds
+ * a constant value to the skew adjustment output.
+ * @param dev 		- The device structure.
+ * @param en	 	- The enable or disable value of SR_INV.
+ * @return    		- Result of the writing procedure, error code otherwise.
+ */
+int adf4377_set_en_sr_inv_adj(struct adf4377_dev *dev, bool en)
+{
+	uint8_t tmp;
+
+	if (!dev)
+		return -EINVAL;
+
+	dev->sr_inv = en;
+
+	tmp = ADF4377_INV_SR(dev->sr_inv);
+	return adf4377_spi_update_bit(dev, 0x43, ADF4377_INV_SR_MSK, tmp);
+}
+
+/**
+ * @brief Gets the value of the set SR_INV_ADJ Adjustment to enable or disable
+ * which adds a constant value to the skew adjustment output.
+ * @param dev 		- The device structure.
+ * @param en	 	- The read value of SR_INV.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4377_get_en_sr_inv_adj(struct adf4377_dev *dev, bool *en)
+{
+	uint8_t tmp;
+	int ret;
+
+	ret = adf4377_spi_read(dev, 0x43, &tmp);
+	if (ret)
+		return ret;
+
+	dev->sr_inv = no_os_field_get(ADF4377_INV_SR_MSK, tmp);
+	*en = dev->sr_inv;
+
+	return 0;
+}
+
+/**
+ * @brief Set enable/disable sysref monitoring.
+ * @param dev 		- The device structure.
+ * @param en	 	- The enable or disable value of the sysref monitor.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4377_set_en_sysref_monitor(struct adf4377_dev *dev, bool en)
+{
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	dev->sysrefout = en;
```

Comment:
don't cache this unless the hardware operation is successful. applies for other functions in this patch too.